task/control-plane-agent/src/mgs_common.rs

```diff
@@ -405,6 +406,72 @@ impl MgsCommon {
             Err(e) => Err(e.into()),
         }
     }
+
+    #[cfg(not(feature = "vpd"))]
+    pub(crate) fn vpd_lock_status_all(
+        &self,
+        _buf: &mut [u8],
+    ) -> Result<usize, SpError> {
+        Err(SpError::Vpd(GatewayVpdError::NotImplemented))
+    }
+
+    #[cfg(feature = "vpd")]
+    pub(crate) fn vpd_lock_status_all(
+        &self,
+        buf: &mut [u8],
+    ) -> Result<usize, SpError> {
+        use task_vpd_api::{Vpd, VpdError};
+        task_slot!(VPD, vpd);
+
+        let vpd = Vpd::from(VPD.get_task_id());
+        let cnt = vpd.num_vpd_devices();
+
+        for (i, entry) in buf.iter_mut().enumerate().take(cnt) {
+            // The API takes a `u8` and we're bounded by the number of devices
+            // in the array which is also going to be less than `u8`
+            let idx = match u8::try_from(i) {
+                Ok(v) => v,
+                Err(_) => unreachable!(),
```

Comment:
Nit: this feels like it should be a `panic!` instead, because it's not immediately obvious that it's unreachable – it's not obvious that `buf.len() <= u8::MAX`, since it's caller-controlled, and `cnt` is a `usize`.

Reply:
One other way to make it clear would be changing `num_vpd_devices` to return a `u8`.

(Also, is `panic!` more expensive than `unreachable!`?)